server/channels/app/permissions_migrations.go

```diff
@@ -1092,6 +1092,25 @@ func (a *App) getProductsBoardsPermissions() (permissionsMap, error) {
 	return transformations, nil
 }
 
+func (a *App) getAddChannelReadContentPermissions() (permissionsMap, error) {
+	t := []permissionTransformation{}
+
+	readChannelContentPermissions := []string{
+		model.PermissionReadChannelContent.Id,
+	}
+
+	t = append(t, permissionTransformation{
+		On: permissionOr(
+			isExactRole(model.SystemAdminRoleId),
+			isExactRole(model.ChannelUserRoleId),
+			isExactRole(model.ChannelGuestRoleId),
+		),
+		Add: readChannelContentPermissions,
+	})
```

Comment:
Are there other roles we should evaluate for changes here?

Reply:
not that I could think of, for all the roles that had `read_channel` this were the only ones that made sense.

Current roles with the `read_channel` permission are:
* System Admin
* Channel User
* Channel Guest
* System Manager
* System read only admin <- I was unsure of this one to be honest
* system user manager